CrudMicroservice.Domain/Services/EuroProtocolService.cs

```diff
@@ -101,5 +109,57 @@ public async Task<IEnumerable<CircumstanceResponseModel>> GetAllCircumstances()
 
             return _mapper.Map<IEnumerable<Circumstance>, IEnumerable<CircumstanceResponseModel>>(circumstances);
         }
+
+        public async Task<EuroProtocolFullResponseModel> GetEuroProtocolBySerialNumber(string serialNumber)
+        {
+            var euroProtocol = await _euroProtocols.GetByFilterAsync(x => x.SerialNumber == serialNumber);
+
+            if (euroProtocol == null)
+            {
+                throw new RestException(HttpStatusCode.NotFound, Resources.ResourceManager.GetString("EuroProtocolNotFound"));
+            }
+
+            var mappedEuroProtocolSimpleResponseModel = _mapper.Map<EuroProtocolSimpleResponseModel>(euroProtocol);
```

Comment:
you should map to EuroProtocolResponseModel, not to EuroProtocolSimpleResponseModel